server/src/main/java/io/crate/metadata/table/TableInfo.java

```diff
@@ -131,4 +132,13 @@ Routing getRouting(ClusterState state,
                        RoutingProvider.ShardSelection shardSelection,
                        SessionContext sessionContext);
 
+    /**
+     * returns the subscription name if the table is used in a Logical Replication subscription.
+     * returns null if this table is not part of a subscription.
+     */
+    @Nullable
+    default String getSubscription() {
```

Comment:
Would it be possible to encapsulate this within the `Operation`? Enum's canot have different states, but maybe we can move from an enum to a sealed class.

I can't put my finger on it, but adding the subscription to every table feels a bit wrong. It creates more coupling between these functionalities.
I am also not sure if this is going to work well once we allow subscriptions on individual partitions.

Reply:
We could also just leave the subscription name out of the error message and we won't have to put this into `TableInfo` at all, but the user would still now that the problem is that this operation is not supported on replicated table.